### PR TITLE
CATROID-677 Make "No internet" message easier to read

### PR DESCRIPTION
**Pull request by blindguardian50**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java [hunk 1]
```diff
@@ -199,6 +199,11 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 		public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
 			if (Utils.checkIsNetworkAvailableAndShowErrorMessage(WebViewActivity.this)) {
 				ToastUtil.showError(getBaseContext(), R.string.error_unknown_error);
+			} else {
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java:202]: actually the old way by showing the toast was completely wrong, cause the error roast will be shown even if the error isn't a network error. for more info please have a look at errors that this callback handle. [webViewErrors](https://developer.android.com/reference/kotlin/android/webkit/WebViewClient#constants)

to indicate no internet access, just compare `errorCode`  or `description` with the corresponding values, which means no need to check if the device has internet access in the first place.

File: catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java [hunk 2]
```diff
@@ -199,6 +199,11 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
...
+				setContentView(R.layout.activity_network_error);
+				setSupportActionBar(findViewById(R.id.toolbar));
+				getSupportActionBar().setIcon(R.drawable.pc_toolbar_icon);
+				getSupportActionBar().setTitle(R.string.app_name);
 			}
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java:207]: in short, please change the body of this callback to suggested :
```suggestion
			if (errorCode == ERROR_HOST_LOOKUP || description.contains("ERR_INTERNET_DISCONNECTED")) {
				setContentView(R.layout.activity_network_error);
				setSupportActionBar(findViewById(R.id.toolbar));
				getSupportActionBar().setIcon(R.drawable.pc_toolbar_icon);
				getSupportActionBar().setTitle(R.string.app_name);
			} else {
				Log.e(TAG, "could'nt connect to the server! info: " + description + " : " + errorCode);
			}
```

File: catroid/src/main/res/layout/activity_network_error.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center_horizontal"
+    android:orientation="vertical">
+
+    <include layout="@layout/toolbar" />
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
```
Comment (ndrnour) [inline catroid/src/main/res/layout/activity_network_error.xml:34]: please change to 
```suggestion
        android:layout_height="wrap_content"
```

**Review by ndrnour — APPROVED** [review]

**Review by wslany — CHANGES_REQUESTED** [review]
Without internet, when the users taps on "Catrobat community" on the main menu, the following screen (which is too dark) still appears:
image
Same also for media gallery.

Tested with build https://jenkins.catrob.at/job/Catroid/job/PR-3739/16/artifact/catroid/build/outputs/apk/catroid/debug/catroid-catroid-debug-PR-3739-16.apk on a Galaxy S20 Ultra with Android 10.

**Comment by blindguardian50** [comment]
This is very strange as i tested https://jenkins.catrob.at/job/Catroid/job/PR-3739/20/artifact/catroid/build/outputs/apk/catroid/debug/catroid-catroid-debug-PR-3739-20.apk because https://jenkins.catrob.at/job/Catroid/job/PR-3739/16/artifact/catroid/build/outputs/apk/catroid/debug/catroid-catroid-debug-PR-3739-16.apk is not available anymore.  When I tested it with the emulator in Android Studio with Android 10 everything seemed to work when the user taps on "Catrobat community" and when the media gallery should be opened too. Do you get the above error message also with other devices or other android versions? Because since I cannot reproduce the error I don't know how to correct it. The intended screen that I always get looks like this:
grafik
but I never get the screen you showed above

**Review by gPathpp — APPROVED** [review]
LGTM!
Please rebase 🙏

**Review by wslany — CHANGES_REQUESTED** [review]
I just tried the following, using browser stack with a Samsung Galaxy S20 Ultra, Android 10, with a simulated unavailable network, and with https://jenkins.catrob.at/job/Catroid/job/PR-3739/36/artifact/catroid/build/outputs/apk/catroid/debug/catroid-catroid-debug-PR-3739-36.apk as the build. 

When I tried to add a new object using the media gallery, the following screen is shown:
image
Trying the leave this screen using the Android back button leads to an ANR and a timeout. 

Same behavior when I tap on the "Catrobat community" on the main screen.

When following the same steps on a Samsung Galaxy S6 with Android 5.0, the same screen is again shown:
image

I also tried it on a Pixel 4 with Android 11, with the same result:
image

I cannot imagine what's wrong. Have you tried with real devices? Thanks.

**Comment by gPathpp** [comment]
> I cannot imagine what's wrong. Have you tried with real devices? Thanks.

Yes, I've tested it with my Android 10 Device (One Plus 6) and multiple Emulators, it works perfectly fine.
It's strange that it doesn't work with those devices. Is it possible for @blindguardian50 to debug on these phones via browser?

**Pull request closed without merging**